django/db/models/fields/tuple_lookups.py

```diff
@@ -0,0 +1,222 @@
+
+from django.core.exceptions import EmptyResultSet
+from django.db.models.expressions import Cols
+from django.db.models.lookups import (
+    Exact,
+    GreaterThan,
+    GreaterThanOrEqual,
+    In,
+    IsNull,
+    LessThan,
+    LessThanOrEqual,
+)
+
+
+class TupleLookupMixin:
+    def get_prep_lookup(self):
+        self.check_tuple_lookup()
+        return super().get_prep_lookup()
+
+    def check_tuple_lookup(self):
+        assert isinstance(self.lhs, Cols)
+        self.check_rhs_is_tuple_or_list()
+        self.check_rhs_length_equals_lhs_length()
+
+    def check_rhs_is_tuple_or_list(self):
+        if not isinstance(self.rhs, (tuple, list)):
+            raise ValueError(
+                f"'{self.lookup_name}' lookup of '{self.lhs.field.name}' field "
+                "must be a tuple or a list"
+            )
+
+    def check_rhs_length_equals_lhs_length(self):
+        if len(self.lhs) != len(self.rhs):
+            raise ValueError(
+                f"'{self.lookup_name}' lookup of '{self.lhs.field.name}' field "
+                f"must have {len(self.lhs)} elements"
+            )
+
+    def check_rhs_is_collection_of_tuples_or_lists(self):
+        if not all(isinstance(vals, (tuple, list)) for vals in self.rhs):
+            raise ValueError(
+                f"'{self.lookup_name}' lookup of '{self.lhs.field.name}' field "
+                f"must be a collection of tuples or lists"
+            )
+
+    def check_rhs_elements_length_equals_lhs_length(self):
+        if not all(len(self.lhs) == len(vals) for vals in self.rhs):
+            raise ValueError(
+                f"'{self.lookup_name}' lookup of '{self.lhs.field.name}' field "
+                f"must have {len(self.lhs)} elements each"
+            )
+
+
+class TupleExact(TupleLookupMixin, Exact):
+    def as_sql(self, compiler, connection):
+        from django.db.models.sql.where import AND, WhereNode
+
+        # e.g.: (a, b, c) == (x, y, z) as SQL:
+        # WHERE a = x AND b = y AND c = z
+        cols = self.lhs.get_cols()
+        lookups = [Exact(col, val) for col, val in zip(cols, self.rhs)]
+        root = WhereNode(lookups, connector=AND)
+
+        return root.as_sql(compiler, connection)
+
+
+class TupleIsNull(IsNull):
+    def as_sql(self, compiler, connection):
+        from django.db.models.sql.where import AND, WhereNode
+
+        # e.g.: (a, b, c) is None as SQL:
+        # WHERE a IS NULL AND b IS NULL AND c IS NULL
+        vals = self.rhs
+        if isinstance(vals, bool):
+            vals = [vals] * len(self.lhs)
+
+        cols = self.lhs.get_cols()
+        lookups = [IsNull(col, val) for col, val in zip(cols, vals)]
+        root = WhereNode(lookups, connector=AND)
+
+        return root.as_sql(compiler, connection)
+
+
+class TupleGreaterThan(TupleLookupMixin, GreaterThan):
+    def as_sql(self, compiler, connection):
+        from django.db.models.sql.where import AND, OR, WhereNode
+
+        # e.g.: (a, b, c) > (x, y, z) as SQL:
```

Comment:
Is this only necessary on Oracle? It seems that `(a, b, c) > (x, y, z)` works on [SQLite](https://dbfiddle.uk/tQuFaMWK), [MySQL](https://dbfiddle.uk/EGXevAkR), and [PostgreSQL](https://dbfiddle.uk/EGXevAkR)?

Reply:
Yes, you're correct. I addressed this in the main branch.